lib/client/db/mcp/server.go

```diff
@@ -0,0 +1,135 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package mcp
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+
+	"github.com/ghodss/yaml"
+	"github.com/gravitational/trace"
+	"github.com/mark3labs/mcp-go/mcp"
+	mcpserver "github.com/mark3labs/mcp-go/server"
+
+	"github.com/gravitational/teleport"
+)
+
+// listDatabasesTool is the MCP tool that list all databases being served
+// (from all protocols).
+var listDatabasesTool = mcp.NewTool(listDatabasesToolName,
+	mcp.WithDescription("List database resources available to be used with Teleport tools."),
+)
+
+// RootServer database access root MCP server.
```

Comment:
You might want to add a little bit more details about how this server is being used. For example, the fact that it runs on the client and what it exposes.

Reply:
I've updated the server godoc.